audittools/backing_store_sql_test.go

```diff
@@ -0,0 +1,289 @@
+		}
+	}
+
+	// Valid table names should be accepted
+	validNames := []string{
+		"audit_events",
+		"AuditEvents",
+		"_audit_events",
+		"audit_events_123",
+	}
+
+	for _, tableName := range validNames {
+		configJSON := fmt.Sprintf(`{"type":"sql","params":{"dsn":%q,"table_name":%q}}`,
+			dsn, tableName)
+
+		store, err := NewBackingStore(configJSON, nil)
+		if err != nil {
+			t.Errorf("expected no error for valid table name %q, got: %v", tableName, err)
+			continue
+		}
+		store.Close()
+
+		// Clean up table
+		db, err := sql.Open("postgres", dsn)
+		if err == nil {
+			_, _ = db.Exec("DROP TABLE IF EXISTS " + tableName) //nolint:errcheck // cleanup in test
+			db.Close()
+		}
+	}
+}
+
+// Test helper types and functions
+
+type SQLBackingStoreOpts struct {
+	DSN        string
+	TableName  string
+	BatchSize  int
+	MaxEvents  int
+	DriverName string
+}
+
+func newTestSQLBackingStore(t *testing.T, opts SQLBackingStoreOpts) *SQLBackingStore {
+	t.Helper()
+
+	if opts.DSN == "" {
+		opts.DSN = getTestDSN(t)
+	}
+
+	if opts.TableName == "" {
+		// Use unique table name per test to allow parallel execution
+		opts.TableName = fmt.Sprintf("audit_events_test_%d", os.Getpid())
+	}
+
+	if opts.DriverName == "" {
+		opts.DriverName = "postgres"
+	}
+
+	configJSON := fmt.Sprintf(`{"type":"sql","params":{"dsn":%q,"table_name":%q,"batch_size":%d,"max_events":%d,"driver_name":%q}}`,
+		opts.DSN, opts.TableName, opts.BatchSize, opts.MaxEvents, opts.DriverName)
+
+	store, err := NewBackingStore(configJSON, nil)
+	if err != nil {
+		t.Fatalf("NewBackingStore failed: %v", err)
+	}
+
+	sqlStore, ok := store.(*SQLBackingStore)
+	if !ok {
+		t.Fatalf("expected *SQLBackingStore, got %T", store)
+	}
+
+	return sqlStore
+}
+
+func cleanupTestSQLBackingStore(t *testing.T, store *SQLBackingStore) {
+	t.Helper()
+
+	// Drop test table
+	if store.db != nil {
+		_, err := store.db.Exec("DROP TABLE IF EXISTS " + store.TableName)
+		if err != nil {
+			t.Logf("failed to drop test table: %v", err)
+		}
+	}
+
+	store.Close()
+}
+
+func mustWriteSQL(t *testing.T, store *SQLBackingStore, event cadf.Event) {
+	t.Helper()
+
+	if err := store.Write(event); err != nil {
+		t.Fatalf("Write failed: %v", err)
+	}
+}
+
+func mustReadBatchSQL(t *testing.T, store *SQLBackingStore) []cadf.Event {
+	t.Helper()
+
+	events, commit, err := store.ReadBatch()
+	if err != nil {
+		t.Fatalf("ReadBatch failed: %v", err)
+	}
+
+	if commit != nil {
+		if err := commit(); err != nil {
+			t.Fatalf("commit failed: %v", err)
+		}
+	}
+
+	return events
+}
+
+func getTestDSN(t *testing.T) string {
+	t.Helper()
+
+	// Check for test database environment variable
+	dsn := os.Getenv("AUDITTOOLS_TEST_DB_DSN")
+	if dsn == "" {
+		t.Skip("AUDITTOOLS_TEST_DB_DSN not set, skipping SQL backing store tests. " +
+			"Set to a PostgreSQL connection string to run these tests, e.g.: " +
+			"postgres://user:password@localhost:5432/testdb?sslmode=disable")
+	}
+
+	return dsn
+}
```

Comment:
Should use WithTestDB and ConnectForTest from go-bits/easypg, to allow those tests to run in the go-makefile-maker CI pipeline.